### PR TITLE
Apply Prettier to markdown files

### PR DESCRIPTION
**Pull request by trask**
The doc team has been using Prettier on the website repo and recently proposed using it on semantic conventions repo also.

This PR is the result of

```
npm install -g prettier
prettier --write "**/*.md"
git checkout -- README.md
git checkout -- instrumentation/jmx-metrics/javaagent/README.md
git checkout -- licenses/licenses.md
```

Future work
- explore the 3 excluded markdown files mentioned above
- apply to other file types
- add automation check in the future.

### CODE REVIEW

File: docs/logger-mdc-instrumentation.md [hunk 1]
```diff
@@ -28,11 +28,11 @@ logs can correlate traces/spans with log statements.
...
-| Log4j 2 | 2.7+                       | [opentelemetry-log4j-context-data-2.17-autoconfigure](../instrumentation/log4j/log4j-context-data/log4j-context-data-2.17/library-autoconfigure) |                                                                                     |
-| Logback | 1.0+                       | [opentelemetry-logback-mdc-1.0](../instrumentation/logback/logback-mdc-1.0/library)  |
+| Library | Auto-instrumented versions | Standalone Library Instrumentation                                                                                                               |
+| ------- | -------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ | --- |
+| Log4j 1 | 1.2+                       |                                                                                                                                                  |
+| Log4j 2 | 2.7+                       | [opentelemetry-log4j-context-data-2.17-autoconfigure](../instrumentation/log4j/log4j-context-data/log4j-context-data-2.17/library-autoconfigure) |     |
```
Comment (mateuszrzeszutek) [inline docs/logger-mdc-instrumentation.md:34]: Looks like we have an extra column here
```suggestion
| ------- | -------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ |
| Log4j 1 | 1.2+                       |                                                                                                                                                  |
| Log4j 2 | 2.7+                       | [opentelemetry-log4j-context-data-2.17-autoconfigure](../instrumentation/log4j/log4j-context-data/log4j-context-data-2.17/library-autoconfigure) |
```

**Review by mateuszrzeszutek — APPROVED** [review]
Nice 👍

**Review by breedx-splk — APPROVED** [review]
Cool, yeah automation will be nice here. Thanks!

File: docs/logger-mdc-instrumentation.md (hunk 1 above)
Comment (trask) [reply docs/logger-mdc-instrumentation.md:34]: oops I missed this, resolved by #9024

**Pull request merged**